### PR TITLE
Added of function to AbstractFObjectPropertyInfo

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added of function to AbstractFObjectPropertyInfo.
2. Added of method to FObjectProperty, Array, Enum, Proxy.
3. Consolidated duplicate FObjectProperty refinements.

### CODE REVIEW

File: src/foam/core/AbstractFObjectPropertyInfo.java [hunk 1]
```diff
@@ -9,7 +9,5 @@
 public abstract class AbstractFObjectPropertyInfo
   extends AbstractObjectPropertyInfo
 {
-  //  public int compareValues(FObject o1, FObject o2) {
-  //    return o1.compareTo(o2);
-  //  }
+  public abstract String of();
```
Comment (kgrgreer) [inline src/foam/core/AbstractFObjectPropertyInfo.java]: I think this would be more useful if it returned ClassInfo instead of String.

Comment (kgrgreer) [inline src/foam/core/AbstractFObjectPropertyInfo.java]: DAO's use getOf(), so you could rename it to that for consistency, but I can live with either.

Comment (kirkeaton) [reply src/foam/core/AbstractFObjectPropertyInfo.java]: How would I resolve to the ClassInfo given the class String?

Comment (kirkeaton) [reply src/foam/core/AbstractFObjectPropertyInfo.java]: Changed it to getOf()

File: src/foam/java/refinements.js [hunk 2]
```diff
@@ -555,6 +573,19 @@ foam.CLASS({
...
+  ],
+
+  methods: [
+    function createJavaPropertyInfo_(cls) {
+      var info = this.SUPER(cls);
+      info.method({
+        name: 'getOf',
+        visibility: 'public',
+        type: 'String',
+        body: 'return "' + (this.of ? this.of.id ? this.of.id : this.of : null) + '";'
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:585]: 'of' is required for Enums, so would just 'this.of'id' be sufficient?

File: src/foam/java/refinements.js [hunk 3]
```diff
@@ -555,6 +573,19 @@ foam.CLASS({
...
     ['javaJSONParser', 'foam.lib.json.FObjectParser']
+  ],
+
+  methods: [
+    function createJavaPropertyInfo_(cls) {
+      var info = this.SUPER(cls);
+      info.method({
+        name: 'getOf',
+        visibility: 'public',
+        type: 'String',
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:584]: Wouldn't ClassInfo be a more useful return type than String?

File: src/foam/java/refinements.js [hunk 4]
```diff
@@ -672,6 +687,13 @@ foam.CLASS({
       var info = this.SUPER(cls);
       var compare = info.getMethod('compare');
       compare.body = this.compareTemplate();
+
+      info.method({
+        name: 'getOf',
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:692]: How/where is this used? Is it needed?

If 'of' is set, shouldn't it change the 'javaType' above?

File: src/foam/java/refinements.js [hunk 5]
```diff
@@ -821,6 +843,19 @@ foam.CLASS({
...
     ['javaJSONParser', 'foam.lib.json.FObjectParser']
+  ],
+
+  methods: [
+    function createJavaPropertyInfo_(cls) {
+      var info = this.SUPER(cls);
+      info.method({
+        name: 'getOf',
+        visibility: 'public',
+        type: 'String',
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:854]: Wouldn't ClassInfo be a more useful return type?

**Pull request closed without merging**